apps/chat/src/components/Common/ApplicationDialog.tsx

```diff
@@ -212,18 +215,33 @@ const ApplicationDialogView: React.FC<Props> = ({
     }
 
     try {
-      const object: Record<string, string> = JSON.parse(data);
+      const object = JSON.parse(data);
+
+      if (typeof object === 'object' && object !== null) {
+        for (const [key, value] of Object.entries(object)) {
+          if (!key.trim()) {
+            return t('Keys should not be empty');
+          }
+
+          if (['boolean', 'number'].includes(typeof value) || value === null) {
+            continue;
+          }
 
-      for (const [key, value] of Object.entries(object)) {
-        if (!key.trim() || !value.trim()) {
-          return t('Keys and Values should not be empty');
+          if (typeof value === 'string' && !value.trim()) {
+            return t('String values should not be empty');
+          }
+
+          if (!['boolean', 'number', 'string', 'null'].includes(typeof value)) {
+            return t(
+              'Values should be a string, number, boolean value, or null',
+            );
+          }
         }
       }
+      return true;
     } catch (error) {
       return t('Invalid JSON string');
     }
-
-    return true;
   };
 
   const onChangeHandlerVersion = (
```

Comment:
also react developers have convention, that functions passed as props called as **on**ChangeSomething. And if function is a part of the components it's better to give a name started with handle, like **handle**ChangeSomething